### PR TITLE
KYLO-1235 Added enableTLS configuration option to the connection to Cloudera Manager

### PR DESCRIPTION
**Pull request by fabiannecci**

### CODE REVIEW

File: plugins/service-monitor-cloudera/service-monitor-cloudera-service/src/test/java/com/thinkbiganalytics/servicemonitor/rest/client/cdh/KyloClouderaClientTest.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    ClouderaClient clouderaClient;
+
+    @Test
+    public void testHttps() {
+        try {
+            Method generateAddress = ClouderaManagerClientBuilder.class.getDeclaredMethod("generateAddress");
+            generateAddress.setAccessible(true);
+            ClouderaManagerClientBuilder clouderaManagerClientBuilder = clouderaClient.getClouderaManagerClientBuilder();
+            String address = (String) generateAddress.invoke(clouderaManagerClientBuilder);
+            Assert.assertEquals(address, "https://cloudera:7183/api/");
```
Comment (KyloTech) [inline plugins/service-monitor-cloudera/service-monitor-cloudera-service/src/test/java/com/thinkbiganalytics/servicemonitor/rest/client/cdh/KyloClouderaClientTest.java]: MAJOR Swap these 2 arguments so they are in the correct order: expected value, actual value. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3415)

File: plugins/service-monitor-cloudera/service-monitor-cloudera-service/src/main/java/com/thinkbiganalytics/servicemonitor/rest/client/RestClientConfig.java [hunk 2]
```diff
@@ -29,6 +29,7 @@
     private String password = "PASSWORD";
```
Comment (KyloTech) [inline plugins/service-monitor-cloudera/service-monitor-cloudera-service/src/main/java/com/thinkbiganalytics/servicemonitor/rest/client/RestClientConfig.java]: BLOCKER 'password' detected in this expression, review this potentially hardcoded credential. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2068)

File: plugins/service-monitor-cloudera/service-monitor-cloudera-service/src/main/java/com/thinkbiganalytics/servicemonitor/rest/client/cdh/ClouderaClient.java [hunk 3]
```diff
@@ -75,10 +76,12 @@ public void setClouderaManagerClientBuilder() {
         Integer port = new Integer(portString);
         String username = clientConfig.getUsername();
         String password = clientConfig.getPassword();
-        LOG.info("Created New Cloudera Client for Host [" + host + ":" + port + "], user: [" + username + "]");
-        this.clouderaManagerClientBuilder =
-            new ClouderaManagerClientBuilder().withHost(host).withPort(port).withUsernamePassword(username, password);
 
+        LOG.info("Created New Cloudera Client for Host [" + host + ":" + port + "], user: [" + username + "]");
```
Comment (KyloTech) [inline plugins/service-monitor-cloudera/service-monitor-cloudera-service/src/main/java/com/thinkbiganalytics/servicemonitor/rest/client/cdh/ClouderaClient.java]: MAJOR Use the built-in formatting to construct this argument. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2629)

**Comment by KyloTech** [comment]
SonarQube analysis reported 6 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![MAJOR][MAJOR] 5 major

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [RestClientConfig.java#L9](https://github.com/Teradata/kylo/blob/f2d8cc2db81b3156a41bc27adf0e8cdf5bc72167/plugins/service-monitor-cloudera/service-monitor-cloudera-service/src/main/java/com/thinkbiganalytics/servicemonitor/rest/client/RestClientConfig.java#L9): This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [ClouderaClient.java#L9](https://github.com/Teradata/kylo/blob/f2d8cc2db81b3156a41bc27adf0e8cdf5bc72167/plugins/service-monitor-cloudera/service-monitor-cloudera-service/src/main/java/com/thinkbiganalytics/servicemonitor/rest/client/cdh/ClouderaClient.java#L9): This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [ClouderaClient.java#L98](https://github.com/Teradata/kylo/blob/f2d8cc2db81b3156a41bc27adf0e8cdf5bc72167/plugins/service-monitor-cloudera/service-monitor-cloudera-service/src/main/java/com/thinkbiganalytics/servicemonitor/rest/client/cdh/ClouderaClient.java#L98): Catch Exception instead of Throwable. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1181)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by freastro** [comment]
Thanks!

**Pull request merged**